doc/ompl_constrained_planning/scripts/ompl_constrained_planning_tutorial.py

```diff
@@ -0,0 +1,736 @@
+
+    path_constraints = moveit_msgs.msg.Constraints()
+    path_constraints.orientation_constraints.append(ocm)
+    pose_goal = tutorial.create_pose_goal_under_obstacle()
+    # pcm = tutorial.create_vertical_plane_constraints()
+
+    move_group.set_start_state(start_state)
+    move_group.set_pose_target(pose_goal)
+    move_group.set_path_constraints(path_constraints)
+    move_group.set_planning_time(1)
+    move_group.plan()
+    move_group.clear_path_constraints()
+
+    print(
+        "============ Press enter to continue with the orientation constrained planning problem with obstacle."
+    )
+    input()
+
+    tutorial.remove_all_markers()
+    tutorial.add_obstacle()
+
+    pose_goal = tutorial.create_pose_goal_under_obstacle()
+    ocm = tutorial.create_orientation_constraints()
+
+    # We need two wrap the constraints in a generic `Constraints` message.
+    path_constraints = moveit_msgs.msg.Constraints()
+    path_constraints.orientation_constraints.append(ocm)
+
+    move_group.set_start_state(start_state)
+    move_group.set_pose_target(pose_goal)
+    move_group.set_path_constraints(path_constraints)
+    move_group.set_planning_time(10)
+    move_group.plan()
+    move_group.clear_path_constraints()
+
+    # print("============ Press enter to continue with the pose constrained planning problem.")
+    # input()
+    # tutorial.remove_all_markers()
+    # tutorial.remove_obstacle()
+
+    # pcm = tutorial.create_vertical_plane_constraints()
+    # ocm = tutorial.create_orientation_constraints()
+
+    # path_constraints = moveit_msgs.msg.Constraints()
+    # path_constraints.orientation_constraints.append(ocm)
+    # path_constraints.position_constraints.append(pcm)
+    # pose_goal = tutorial.create_pose_goal_under_obstacle()
+    # # pcm = tutorial.create_vertical_plane_constraints()
+
+    # move_group.set_start_state(start_state)
+    # move_group.set_pose_target(pose_goal)
+    # move_group.set_path_constraints(path_constraints)
+    # move_group.set_planning_time(1)
+    # move_group.plan()
+    # move_group.clear_path_constraints()
+
+    # print("============ Press enter to continue with the pose constrained planning problem with obstacle.")
+    # input()
+    # tutorial.remove_all_markers()
+    # tutorial.add_obstacle()
+
+    # pcm = tutorial.create_vertical_plane_constraints()
+    # ocm = tutorial.create_orientation_constraints()
+
+    # path_constraints = moveit_msgs.msg.Constraints()
+    # path_constraints.orientation_constraints.append(ocm)
+    # path_constraints.position_constraints.append(pcm)
+    # pose_goal = tutorial.create_pose_goal_under_obstacle()
+    # # pcm = tutorial.create_vertical_plane_constraints()
+
+    # move_group.set_start_state(start_state)
+    # move_group.set_pose_target(pose_goal)
+    # move_group.set_path_constraints(path_constraints)
+    # move_group.set_planning_time(60)
+    # move_group.plan()
+    # move_group.clear_path_constraints()
```

Comment:
If this is replaced by a new version in the tutorial we can remove the code in comments to avoid confusion. I forgot why this is in comments.

Reply:
I added and could test it. I had in mind to maybe add pose constraints to my moveit PR. I think if we keep orientation constraints in this PR here we can keep pose constraints too. I will wait to see how we arrange the moveit PRs before resolving this.

Reply:
I would suggest keeping this PR open until orientation constraints are also merged. So indeed it could be interesting to add this example for real.

Also, I noticed some of the tutorial cases still fail sometimes. (Planner timeout I think?) @gautz if you have ideas to improve this situation they're more than welcome :)

Reply:
Do you plan to work on an orientation constraints PR ? Otherwise I could try to cherry-pick relevant commits and open a PR this week.

Solving also fails for collision-avoidance examples around 1/4th of the time. My plan was to test Atlas based solving to see whether it improves performance and reliability. I don't have more ideas, except maybe tuning some numerical parameters.